brand/admin.py

```diff
@@ -21,18 +22,62 @@
 from brand.models.embrace_campaign import EmbraceCampaign
 from datasource.constants import model_names
 from datasource.models.datasource import Datasource, SuggestedAssociation
-
-from .models import Brand, Commentary
+from django.contrib.admin.widgets import ForeignKeyRawIdWidget
+from .models import Brand, Commentary, Contact
 
 from django.core.exceptions import ObjectDoesNotExist
 from brand.forms import EmbraceCampaignForm
+from nested_admin import NestedStackedInline, NestedModelAdmin
+from django import forms
+from taggit.forms import TagWidget
+
+
+class ContactForm(forms.ModelForm):
+    class Meta:
+        model = Contact
+        fields = ["email"]
+        widgets = {"email": forms.SelectMultiple()}
+
+    def __init__(self, *args, **kwargs):
+        super(ContactForm, self).__init__(*args, **kwargs)
+        if self.instance.pk is None:
+            self.fields["email"] = forms.ModelMultipleChoiceField(
+                queryset=Contact.objects.filter(commentary__isnull=True),
+                widget=forms.SelectMultiple,
+                label="emails:",
+            )
+        else:
+            self.fields["email"] = forms.ModelMultipleChoiceField(
+                queryset=Contact.objects.filter(
+                    commentary__brand__id=self.instance.commentary.brand.id
+                ),
+                widget=forms.SelectMultiple,
+                label="emails:",
+            )
+
+
+class ContactInline(NestedStackedInline):
```

Comment:
What you did here is nice but I actually think it's overkill. You can easily get away with just linking the contact details or displaying a big message like "no contact details linked, why don't you add one" message if there aren't any. It would be easier than showing that really difficult multi-select box and overriding the permissions via the admin.

Reply:
The point of these thing is really maintainability over ease of use, so I'd rather have you just use a `@admin.display` wrapper for a read only thing than have to spend so much time manipulating the form displays